optimism/inverse/MechanicsInverse.py

```diff
@@ -0,0 +1,103 @@
+from collections import namedtuple
+
+from optimism.JaxConfig import *
+from optimism import FunctionSpace
+from optimism import Interpolants
+from optimism.TensorMath import tensor_2D_to_3D
+
+IvsUpdateInverseFunctions = namedtuple('IvsUpdateInverseFunctions',
+                                      ['ivs_update_jac_ivs_prev',
+                                       'ivs_update_jac_disp_vjp',
+                                       'ivs_update_jac_coords_vjp'])
+
+PathDependentResidualInverseFunctions = namedtuple('PathDependentResidualInverseFunctions',
+                                                  ['residual_jac_ivs_prev_vjp',
+                                                   'residual_jac_coords_vjp'])
+
+ResidualInverseFunctions = namedtuple('ResidualInverseFunctions',
+                                     ['residual_jac_coords_vjp'])
+
+def _compute_quadrature_point_field_gradient(u, shapeGrad):
```

Comment:
This function appears to be copied and pasted from `FunctionSpace` - why not use that directly?

Reply:
Good catch, fixed it.